### ISSUE TITLE
RunInstances API call, attempt multiple instance types

### ISSUE THREAD

**Issue by symonds** [AutoSpotting/AutoSpotting#263]
## Issue type ##

- Feature Idea

## Summary ##

When using the new RunInstances API call (https://github.com/cristim/autospotting/pull/257) and there is high demand on the instance type AWS responds with `InsufficientInstanceCapacity`

Which Autospotting returns

```
2018/06/07 11:56:39 instance.go:306: Couldn't launch spot instance: InsufficientInstanceCapacity: There is no Spot capacity available that matches your request.
	status code: 500, request id: 3045e3ba-c3f0-40d1-903b-92327ec3b5c1
2018/06/07 11:56:39 autoscaling.go:337: Could not launch cheapest spot instance: InsufficientInstanceCapacity: There is no Spot capacity available that matches your request.
	status code: 500, request id: 3045e3ba-c3f0-40d1-903b-92327ec3b5c1
```

It would be nice if Autospotting went back to `getCheapestCompatibleSpotInstanceType` with the failed instance type appended to `disallowedList` and attempted to bid for the next cheapest instance.

I would limit this action to 3 attempts.

**Comment by cristim** [comment]
This is a great proposal, thanks!

**Comment by cristim** [comment]
I'm thinking to just iterate over the list of compatible instance types ordered increasingly by price and launch them one by one until we get one that can be launched successfully.

**Comment by sunnynew** [comment]
Hi @cristim 

First of all congrats on this wonderful tool. 

I wanted to know about this particular issue, anytime soon can we expect a fix.  As you mentioned just make Autospotting to iterate over all instance type mentioned in allowed_instance_types list. 

This will be a lifesaver. 

Thanks!

**Comment by cristim** [comment]
@sunnynew thanks for your feedback, I appreciate it.

I'm now focusing on #294 so I'm not planning to work on this in the very near future, but feel free to take it over and contribute a PR if you need this so badly and have some time to spare.

I can help you with guidance if you need any.

**Comment by cristim** [comment]
Fixed in #306
